graphql_compiler/macros/__init__.py

```diff
@@ -167,16 +170,26 @@ def get_schema_for_macro_definition(schema):
     """Returns a schema with macro directives.
 
     This returned schema can be used to validate macro definitions, and support GraphQL
-    macro editors, enabling them to autocomplete one the @macro_edge_definition and
-    @macro_edge_target directives.
+    macro editors, enabling them to autocomplete on the @macro_edge_definition and
+    @macro_edge_target directives. Some directives that are disallowed in macro edge definitions,
+    like @output and @output_source, will be removed from the directives list.
 
     Args:
         schema: GraphQLSchema over which we want to write macros
 
     Returns:
         GraphQLSchema usable for writing macros
     """
-    raise NotImplementedError()  # TODO(bojanserafimov): Implement
+    macro_definition_schema = copy(schema)
```

Comment:
The copying seems fine to me. I don't understand what you mean by vulnerabilities.

Reply:
The returned schema's fields are the same copy as the argument schema's fields. So if a macro editor edits the schema returned from this function, we might run into issues. How can we make the fields immutable?